File: Acwing/cpp/基础算法Partial/799.cpp
#include<iostream>
#include<vector>

using namespace std;

int main(){
    int n;
    std::cin >> n;
    vector<int> a(n, 0);
    vector<int> cnt(100002, 0);
    int j = 0;
    int ans = 0;
    for(int i=0; i<n; i++){
        std::cin >> a[i];
        cnt[a[i]]++;
        while(j <= i && cnt[a[i]] > 1){
            if(cnt[a[i]] > 1){
                cnt[a[j]]--;
                j++;
            }
        }
        ans = ans > (i - j + 1) ? ans : (i - j + 1);
    }
    std::cout << ans << std::endl;
    return 0;
}

File: Acwing/py/基础算法Partial/799.py
n = int(input().strip())

a = list(map(int, input().strip().split()))
cnt = [0 for _ in range(100002)]

j = 0
ans = 0

for i in range(n):
    cnt[a[i]] += 1
    while j <= i and cnt[a[i]] > 1:
        cnt[a[j]] -= 1
        j += 1
    ans = ans if ans > (i - j + 1) else (i - j + 1)

print(ans)